products/sle/main.pm

```diff
@@ -689,8 +700,12 @@ sub load_inst_tests {
         if (get_var('PATTERNS') || get_var('PACKAGES')) {
             loadtest "installation/installation_overview_before";
             loadtest "installation/select_patterns_and_packages";
-        }    # With SLE15 we change desktop using role and not by inselecting packages (Use SYSTEM_ROLE variable)
-        elsif (!check_var('DESKTOP', default_desktop) && !sle_version_at_least('15')) {
+        }
+        elsif (!check_var('DESKTOP', default_desktop)
+            && (!sle_version_at_least('15') || check_var('DESKTOP', 'minimalx')))
+        {
```

Comment:
Attempting a sles4sap textmode installation like [this one](https://openqa.suse.de/tests/1405848) with the new code still gives me some problems:

1. Setting `DESKTOP=textmode` and not setting `SYSTEM_ROLE` produces a failure in `first_boot`, because the test expects a text-based linux login needle due to the value of `DESKTOP`, but instead gets a DM login screen because the desktop isn't changed during installation (`SYSTEM_ROLE` defaults to `default` and `installation/change_desktop` is not loaded due to `!sle_version_at_least('15')`)

2. Setting `DESKTOP=textmode` and `SYSTEM_ROLE=textmode` also fails, because the system role introduces an additional step to set up a user account. This can be fixed by adding a `or !check_var('SYSTEM_ROLE', 'default')` on line 689. This will also require a needle for `user_settings` to assert `ROOTONLY=1` in text mode. I feel this is the way to go.

3. Additionaly, the change in `SYSTEM_ROLE` from the default to `textmode` in sles4sap, causes some SAP packages to not be installed, so some `sles4sap/*` tests fail; I'll modify those to take this changes into consideration.